docs/source/API/core/printf.rst

```diff
@@ -0,0 +1,17 @@
+``Kokkos::printf``
+=================
+
+.. role:: cppkokkos(code)
+    :language: cppkokkos
+
+Defined in header ``<Kokkos_Core.hpp>``
+
+.. code-block:: cpp
+
+    template <typename... Args>
+    KOKKOS_FUNCTION int printf(const char* format, Args... args);
+
+For the Serial backend, ``Kokkos::printf`` is implemented in terms of ``std::printf`` printing the data specified in ``format`` and ``args...`` to ``stdout``.
+Returns the number of characters written if successful or a negative value if an error encountered.
```

Comment:
A few questions:

1) for serial implementations, `Kokkos::printf` just calls `std::printf`, and it is only in this case that formatted arguments will be printed to `stdout`. while in all other non-serial cases, you'll only have an `int` to `stdout`?

2) By "successful" vs. "error encountered," do you mean in the print statement itself?

3) For the non-serial cases, how useful is it to have a print statement return an `int` representing the characters that were written?  I'm assuming there's some difficulty in returning the characters themselves in the non-serial cases .... ?

Reply:
1. No, they typically all print to stdout in the same way.
2. Yes, if, e.g., there is an encoding error, `std::printf` would return a negative value instead of the numbers of characters written. It's not quite clear what all of the device implementations would do in that (edge) case.
3. We could just not return anything but the idea is to wrap `std::printf` on the host and make a best effort to be close to that on the device.